src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -0,0 +1,260 @@
+                    receiverType,
+                    TypeCompareKind.ConsiderEverything,
+                    ref useSiteDiagnostics))
+                {
+                    diagnostics.Add(
+                        ErrorCode.ERR_ContainingTypeMustDeriveFromWithReturnType,
+                        syntax.Receiver.Location,
+                        receiverType,
+                        withMethod.ReturnType);
+                }
+
+                useSiteDiagnostics = null;
+
+                // Build WithMethod member list
+                var matchingMembers = ArrayBuilder<Symbol?>.GetInstance(withMethod.ParameterCount);
+                foreach (var p in withMethod.Parameters)
+                {
+                    this.LookupMembersInType(
+                        lookupResult,
+                        receiverType,
+                        p.Name,
+                        arity: 0,
+                        basesBeingResolved: null,
+                        options: LookupOptions.MustBeInstance,
+                        originalBinder: this,
+                        diagnose: false,
+                        useSiteDiagnostics: ref useSiteDiagnostics);
+
+                    Symbol? member;
+                    if (!lookupResult.IsSingleViable ||
+                        (lookupResult.SingleSymbolOrDefault is var symbol &&
+                         symbol.Kind != SymbolKind.Field &&
+                         symbol.Kind != SymbolKind.Property))
+                    {
+                        diagnostics.Add(
+                            ErrorCode.ERR_WithParameterWithoutMatchingMember,
+                            syntax.Receiver.Location,
+                            receiverType,
+                            p.Name);
+                        member = null;
+                    }
+                    else
+                    {
+                        member = lookupResult.SingleSymbolOrDefault;
+                        if (!member.GetTypeOrReturnType().Equals(
+                                p.TypeWithAnnotations,
+                                TypeCompareKind.ConsiderEverything))
+                        {
+                            diagnostics.Add(
+                                ErrorCode.ERR_WithParameterTypeDoesntMatchMemberType,
+                                syntax.Receiver.Location,
+                                p.Name,
+                                p.Type,
+                                member.GetTypeOrReturnType());
+                        }
+                    }
+
+                    lookupResult.Clear();
+                    useSiteDiagnostics = null;
+                    matchingMembers.Add(member);
+                }
+                withMembers = matchingMembers.ToImmutableAndFree();
+            }
+
+            // Verify that the member name exists in the 'With' method parameter list
+            // and that there is an implicit conversion from the expression to the
```

Comment:
>there is an implicit conversion [](start = 24, length = 31)

The code below does not appear to do this.

Reply:
Fixed the comment. The conversion needs to be placed in the expression, so I moved it up in the code.